lib/ansible/_internal/_ansiballz.py

```diff
@@ -0,0 +1,265 @@
+    profile: str,
+    plugin_info_dict: dict[str, object],
+    date_time: datetime.datetime,
+    coverage_config: str | None,
+    coverage_output: str | None,
+    rlimit_nofile: int,
+) -> None:
+    import os
+    import os.path
+
+    # Access to the working directory is required by Python when using pipelining, as well as for the coverage module.
+    # Some platforms, such as macOS, may not allow querying the working directory when using become to drop privileges.
+    try:
+        os.getcwd()
+    except OSError:
+        try:
+            os.chdir(os.path.expanduser('~'))
+        except OSError:
+            os.chdir('/')
+
+    if rlimit_nofile:
+        import resource
+
+        existing_soft, existing_hard = resource.getrlimit(resource.RLIMIT_NOFILE)
+
+        # adjust soft limit subject to existing hard limit
+        requested_soft = min(existing_hard, rlimit_nofile)
+
+        if requested_soft != existing_soft:
+            try:
+                resource.setrlimit(resource.RLIMIT_NOFILE, (requested_soft, existing_hard))
+            except ValueError:
+                # some platforms (eg macOS) lie about their hard limit
+                pass
+
+    import sys
+    import __main__
+
+    # For some distros and python versions we pick up this script in the temporary
+    # directory.  This leads to problems when the ansible module masks a python
+    # library that another import needs.  We have not figured out what about the
+    # specific distros and python versions causes this to behave differently.
+    #
+    # Tested distros:
+    # Fedora23 with python3.4  Works
+    # Ubuntu15.10 with python2.7  Works
+    # Ubuntu15.10 with python3.4  Fails without this
+    # Ubuntu16.04.1 with python3.5  Fails without this
+    # To test on another platform:
+    # * use the copy module (since this shadows the stdlib copy module)
+    # * Turn off pipelining
+    # * Make sure that the destination file does not exist
+    # * ansible ubuntu16-test -m copy -a 'src=/etc/motd dest=/var/tmp/m'
+    # This will traceback in shutil.  Looking at the complete traceback will show
+    # that shutil is importing copy which finds the ansible module instead of the
+    # stdlib module
+    scriptdir = None
+    try:
+        scriptdir = os.path.dirname(os.path.realpath(__main__.__file__))
+    except (AttributeError, OSError):
+        # Some platforms don't set __file__ when reading from stdin
+        # OSX raises OSError if using abspath() in a directory we don't have
+        # permission to read (realpath calls abspath)
+        pass
+
+    # Strip cwd from sys.path to avoid potential permissions issues
+    excludes = {'', '.', scriptdir}
+    sys.path = [p for p in sys.path if p not in excludes]
+
+    import base64
+    import shutil
+    import tempfile
+    import zipfile
+
+    def invoke_module(modlib_path: str, json_params: bytes) -> None:
+        # When installed via setuptools (including python setup.py install),
```

Comment:
Is this even possible anymore, now that we don't actually have a `setup.py`?

Reply:
not ours, but other people can install modules in diff ways

Reply:
Various comments in this module are out-of-date. The code was relocated with a few modifications, but we didn't take the time to rewrite the entire module. The original was here:

https://github.com/ansible/ansible/blob/183c695be1312ac1d4829a990ab3cd5d1e4ca5a5/lib/ansible/executor/module_common.py#L174